consent-ws/src/main/java/org/broadinstitute/consent/http/service/ontologyIndexer/IndexOntologyService.java

```diff
@@ -47,12 +51,12 @@ public String getIndexName() {
      * @throws IOException The exception
      */
     public void indexOntologies(List<StreamRec> streamRecList) throws IOException {
-
-        utils.validateIndexExists(client, indexName);
-
-        try {
+        try(RestClient client = getRestClient()) {
             for (StreamRec streamRec : streamRecList) {
 
+                // Deprecate everything that might already exist for this ontology file
+                utils.bulkDeprecateTerms(client, indexName, streamRec.getOntologyType());
+
```

Comment:
At present, we don't have a way to distinguish between the two cases (existing terms and new terms) after posting them to ES. The previous code would leave in place old terms that were not in the new file which was definitely a bug. Also, since we're just deprecating terms (by setting a field to `false`), we can always fix things up after the fact if there is a problem with the file.